src/MobileUI/ViewModels/AddLinkedInViewModel.cs

```diff
@@ -0,0 +1,118 @@
+using CommunityToolkit.Mvvm.ComponentModel;
+using CommunityToolkit.Mvvm.Input;
+using Mopups.Services;
+using SSW.Rewards.Mobile.Controls;
+
+namespace SSW.Rewards.Mobile.ViewModels;
+
+public partial class AddLinkedInViewModel : BaseViewModel
+{
+    private readonly IUserService _userService;
+    private readonly ISnackbarService _snackbarService;
+
+    [ObservableProperty]
+    private bool _isOverlayVisible;
+
+    [ObservableProperty]
+    private string _inputText;
+
+    [ObservableProperty]
+    private int _cursorPosition;
+
+    [ObservableProperty]
+    private bool _isError;
+
+    [ObservableProperty]
+    private string _errorText;
+
+    public AddLinkedInViewModel(IUserService userService, ISnackbarService snackbarService)
+    {
+        _userService = userService;
+        _snackbarService = snackbarService;
+    }
+
+    [RelayCommand]
+    private async Task Connect()
+    {
+        IsError = false;
+        if (string.IsNullOrWhiteSpace(InputText))
+        {
+            IsError = true;
+            ErrorText = "URL cannot be empty";
+            return;
+        }
+
+        if (!IsUrlValid())
+        {
+            IsError = true;
+            ErrorText = "The URL is not valid";
+            return;
+        }
+
+        await AddLinkedInProfile();
+    }
+
+    [RelayCommand]
+    private async Task ClosePage()
+    {
+        IsOverlayVisible = false;
+        await MopupService.Instance.PopAsync();
+    }
+
+    [RelayCommand]
+    private void InputFocused()
+    {
+        if (string.IsNullOrWhiteSpace(InputText))
+        {
+            InputText = "https://linkedin.com/in/";
+        }
+
+        CursorPosition = InputText.Length; // For some reason, works only on Android
+    }
+
+    private bool IsUrlValid()
+    {
+        var urlParts = InputText.Split('/');
+        var isValid = urlParts.Length == 5 &&
```

Comment:
This could bite us...hopefully the placeholder is sufficient guidance for users

Reply:
It's also pre-filled :)